### PR TITLE
Use PSR4 autoload system by composer

### PR DESCRIPTION
**Pull request by ovr**
- [x] PSR4 by Composer
- [x] Use phpunit by composer
- [x] TravisCI - run phpunit with debug and composer install

### CODE REVIEW

**Comment by ovr** [comment]
ping @nikic 
It's needed for next scrutinizer ci, because i am going to setup "sending code coverage report" to it

File: .travis.yml [hunk 1]
```diff
@@ -11,3 +11,10 @@ matrix:
   allow_failures:
     - php: 7.0
   fast_finish: true
+
+install:
+  - composer self-update
```
Comment (GrahamCampbell) [inline .travis.yml]: :-1:

File: .travis.yml [hunk 2]
```diff
@@ -11,3 +11,10 @@ matrix:
...
+  - composer --prefer-source install
```
Comment (GrahamCampbell) [inline .travis.yml]: `composer install --prefer-source`

File: .travis.yml [hunk 3]
```diff
@@ -11,3 +11,10 @@ matrix:
...
+
+script:
+  - ./vendor/bin/phpunit -v --debug
```
Comment (GrahamCampbell) [inline .travis.yml]: missing newline

File: .travis.yml (hunk 1 above)
Comment (fezfez) [inline .travis.yml]: @GrahamCampbell why ?

Comment (GrahamCampbell) [inline .travis.yml]: why add it?

Comment (GrahamCampbell) [inline .travis.yml]: just wastes time

**Comment by ovr** [comment]
@GrahamCampbell 
Omg, schoolchild, go away with this spaming atack

**Comment by AveVlad** [comment]
> Omg, schoolchild, go away with this spaming atack

:+1: @GrahamCampbell  go contirube laravel

File: .travis.yml (hunk 1 above)
Comment (fezfez) [inline .travis.yml]: Until composer does not have a stable release, its need. By the way its pretty fast

File: .travis.yml (hunk 2 above)
Comment (ovr) [inline .travis.yml]: @GrahamCampbell 
It's not a standart to write parameter in the end
For example, On OSX Just try to use cp with `-R parameter` in the of exec comand

File: .travis.yml [hunk 4]
```diff
@@ -11,3 +11,11 @@ matrix:
...
+
```
Comment (GrahamCampbell) [inline .travis.yml]: too many newlines

File: .travis.yml (hunk 2 above)
Comment (GrahamCampbell) [inline .travis.yml]: it is pretty standard for symfony, which is what composer uses

File: .travis.yml (hunk 1 above)
Comment (GrahamCampbell) [inline .travis.yml]: It's not needed though. If you removed it, everything would still work.

**Comment by GrahamCampbell** [comment]
> Omg, schoolchild, go away with this spaming atack

No need to be rude. The way I see it, I'm preventing bad changes being made to a project I care about.

File: .travis.yml (hunk 4 above)
Comment (fezfez) [inline .travis.yml]: what do mean ?

Comment (GrahamCampbell) [inline .travis.yml]: you've added 2 newline characters instead of 1

Comment (ovr) [inline .travis.yml]: 1 new line in the of file, what is wrong?

File: .travis.yml (hunk 2 above)
Comment (ovr) [inline .travis.yml]: > pretty standard for symfony
Where did you read it?

File: .travis.yml (hunk 1 above)
Comment (ovr) [inline .travis.yml]: @GrahamCampbell 
Sometimes ago I got fails without it
It's not slow

Comment (GrahamCampbell) [inline .travis.yml]: meh, can we at least move it to "before-install" then

Comment (fezfez) [inline .travis.yml]: @GrahamCampbell : Good solution

**Comment by nikic** [comment]
Switched to the PSR-4 autoloader in https://github.com/nikic/PHP-Parser/commit/e3a93561780f750489aa3a1cf99d23dcfa44520f. We may need to switch back to a custom one later if there's going to be aliases again, but I guess it's better to use the generated one for now.

File: .travis.yml (hunk 1 above)
Comment (ovr) [inline .travis.yml]: Guys, it's rly useless but it's needed 1 line for before install and 1 line empty line after, 2 uneeded lines

**Comment by nikic** [comment]
Phpunit dep in composer is now present as well: https://github.com/nikic/PHP-Parser/commit/6f3fd7834a60e176d7734b786cfd1df6b764ed5e

**Pull request closed without merging**